google/cloud/aiplatform/training_jobs.py

```diff
@@ -281,46 +333,474 @@ def package_and_copy_to_gcs(
+        self._model_serving_container_predict_route = (
+            model_serving_container_predict_route
+        )
+        self._model_serving_container_health_route = (
+            model_serving_container_health_route
+        )
+        self._gca_resource = None
+
+    # TODO(b/172365904) add filter split, training_pipline.FilterSplit
+    # TODO(b/172366411) predefined filter split training_pipeline.PredfinedFilterSplit
+    # TODO(b/172368070) add timestamp split, training_pipeline.TimestampSplit
     def run(
         self,
-        dataset: Optional[datasets.Dataset],
-        output_dir: Optional[str] = None,
+        dataset: Optional[datasets.Dataset] = None,
+        model_display_name: Optional[str] = None,
+        base_output_dir: Optional[str] = None,
         args: Optional[Dict[str, str]] = None,
-        replica_count=0,
-        machine_type="n1-standard-4",
-        accelerator_type=None,
-        accelerator_count=0,
-        spec_yaml_path: Optional[str] = None,
-    ) -> models.Model:
-        pass
+        replica_count: str = 0,
+        machine_type: str = "n1-standard-4",
+        accelerator_type: str = "ACCELERATOR_TYPE_UNSPECIFIED",
+        accelerator_count: int = 0,
+        training_fraction_split: float = 0.8,
+        validation_fraction_split: float = 0.1,
+        test_fraction_split: float = 0.1,
+    ) -> Optional[models.Model]:
+        """Runs the custom training job.
+
+        Data fraction splits:
+        Any of ``training_fraction_split``, ``validation_fraction_split`` and
+        ``test_fraction_split`` may optionally be provided, they must sum to up to 1. If
+        the provided ones sum to less than 1, the remainder is assigned to sets as
+        decided by AI Platform.If none of the fractions are set, by default roughly 80%
+        of data will be used for training, 10% for validation, and 10% for test.
 
+        Args:
+            dataset (aiplatform.Dataset):
+                AI Platform to fit this training against. Custom training script should
+                retrieve datasets through passed in environement variables uris:
+
+                os.environ["AIP_TRAINING_DATA_URI"]
+                os.environ["AIP_VALIDATION_DATA_URI"]
+                os.environ["AIP_TEST_DATA_URI"]
+
+                Additionally the dataset format is passed in as:
+
+                os.environ["AIP_DATA_FORMAT"]
+            model_display_name (str):
+                If the script produces a managed AI Platform Model. The display name of
+                the Model. The name can be up to 128 characters long and can be consist
+                of any UTF-8 characters.
+            base_output_dir (str):
+                GCS output directory of job. If not provided a
+                timestamped directory in the staging directory will be used.
+            args (Dict[str, str]):
+                Key-value pair or command line arguments to be passed to the Python
```

Comment:
Maybe some more explanations here, since `format_args` does something implicitly.

Reply:
Changed to args and accept a list to support positional args and flags.